geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -1077,7 +1081,10 @@ protected Set<Object> processReceivedRVV(RegionVersionVector remoteRVV,
         if (id == null) {
           id = myId;
         }
-        if (!remoteRVV.contains(id, stamp.getRegionVersion())) {
+        if (!isPersistentRegion) {
+          foundIds.add(id);
+        }
+        if (isPersistentRegion && !remoteRVV.contains(id, stamp.getRegionVersion())) {
```

Comment:
instead of testing "isPersistentRegion" in this "if" just add an "else" before the "if".